Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth
# Class Mentoring Rework

## Changelog:
- You can ask characters to be class mentors of other characters like you children. Some mentors would like not to take students because of their traits or because of hostility.

## Developer changelog:
- The correspondence between a class mentor and student takes more time.
- Added a fallback/cleanup event in case of death of a class mentor or student during their correspondence.
- Rewrote the localization of the mentoring events (the correspondence, decision description etc.).

## Script changelog (ignore if you can't read CK2 script):
- Added `ai_chance` in `WCCLS.1500` to make sure the AI prefers to pick a class, not cancel the event.
- Removed `opinion_refused_to_be_mentor` from `WCCLS.1500` event.
- Made changes in the Hidden Class mechanics. Added `this_see_prev_class_trait_trigger` trigger.
- Renamed `opinion_rejected_request_for_mentoring` into `opinion_refused_to_be_mentor` because the name was too long.

## How to Test
Try the new features:
- Try to assign mentors to your children.
- Try to select mentors for yourself.
- Make sure the event, where an Educator trains your children when they reach 12 years old, works fine.  You may find 11 years old, assign an Educator with high class level to her/him, wait for the event.
- Make sure the Hidden Class mechanics work. You may test it with mages in the Kaldorei Empire. They start with banned Arcane magic, so mages shouldn't be visible. If you're not mage yourself, you shoudn't see their mana, class trait. You shouldn't be able to `Ask to be Mentor` as well. Though you should be able to `Ask to be Mentor` if your religion is... Sunwell... for example. `religion cult_of_sunwell` in the console to change your religion to Sunwell.

After the mentor has agreed to train you (or your children), make sure `ask_to_be_mentor_ongoing_flag` flag is cleared on you (or your children).
